src/coreclr/jit/llvmcodegen.cpp

```diff
@@ -1722,13 +1706,13 @@ void Llvm::storeObjAtAddress(Value* baseAddress, Value* data, StructDesc* struct
             {
                 // We can't be sure the address is on the heap, it could be the result of pointer arithmetic on a local var.
                 emitHelperCall(CORINFO_HELP_CHECKED_ASSIGN_REF,
-                               {address, castIfNecessary(fieldData, Type::getInt8PtrTy(_llvmContext))});
+                               {address, castIfNecessary(fieldData, llvm::PointerType::getUnqual(_llvmContext))});
```

Comment:
Hmm, when is `fieldData` not a pointer here?

Reply:
E.g. Its the `RhpCheckedAssignRef` for `S_P_TypeLoader_Internal_TypeSystem_Instantiation___cctor` :

```
Generating:                [000043] ------------        t43 =    LCL_VAR   int    V01 tmp1         u:1 (last use)
                                                              /--*  t43    int
Generating: N005 (  3,  2) [000007] n-----------         t7 = *  OBJ       struct<Internal.TypeSystem.Instantiation, 4>
  %16 = load i32, ptr %0, align 4, !dbg !5
                                                              /--*  t10    byref
                                                              +--*  t7     struct
Generating: N004 ( 22,  8) [000012] -ACXG-------              *  STORE_OBJ struct<Internal.TypeSystem.Instantiation, 4> (copy)
```

`Instantiation` is 
```
    public struct Instantiation : IEquatable<Instantiation>
    {
        private TypeDesc[] _genericParameters;
```
So is this because of the LLVM struct representation of an `i32` I guess

Reply:
I see. Bit unexpected (one would think the struct would be normalized to `ptr`), but also pre-existing so nothing to do.